crates/router/src/connector/utils.rs

```diff
@@ -2901,14 +2904,55 @@ pub fn get_sync_integrity_object<T>(
     amount_convertor: &dyn AmountConvertor<Output = T>,
     amount: T,
     currency: String,
+    capture_amount: Option<T>,
 ) -> Result<SyncIntegrityObject, error_stack::Report<errors::ConnectorError>> {
     let currency_enum = enums::Currency::from_str(currency.to_uppercase().as_str())
         .change_context(errors::ConnectorError::ParsingFailed)?;
     let amount_in_minor_unit =
         convert_back_amount_to_minor_units(amount_convertor, amount, currency_enum)?;
 
+    let capture_amount_in_minor_unit = capture_amount
+        .map(|amount| convert_back_amount_to_minor_units(amount_convertor, amount, currency_enum))
+        .transpose()?;
+
     Ok(SyncIntegrityObject {
         amount: Some(amount_in_minor_unit),
         currency: Some(currency_enum),
+        capture_amount: capture_amount_in_minor_unit,
+    })
+}
+
+pub fn get_capture_integrity_object<T>(
+    amount_convertor: &dyn AmountConvertor<Output = T>,
+    capture_amount: Option<T>,
+    currency: String,
+) -> Result<CaptureIntegrityObject, error_stack::Report<errors::ConnectorError>> {
+    let currency_enum = enums::Currency::from_str(currency.to_uppercase().as_str())
+        .change_context(errors::ConnectorError::ParsingFailed)?;
```

Comment:
If this check fails we will throw `internal_server_error`, and we won't be able to recover the payment without directly altering the DB

Reply:
correct, but this parsing failure should only happen if the currency is not supported by our application and we get that in response, Yes! we wont be able to recover this payment since its a data mismatched issue at connector end! it should be 5xx since if the issue is in hyperswitch end then we shouldn't have let the payment go through in the beginning of the request.